mlir/lib/Quantum/Transforms/MergeRotationsPatterns.cpp

```diff
@@ -98,14 +124,33 @@ struct MergeMultiRZRewritePattern : public mlir::OpRewritePattern<MultiRZOp> {
         ValueRange parentInCtrlQubits = parentOp.getInCtrlQubits();
         ValueRange parentInCtrlValues = parentOp.getInCtrlValues();
 
+        ArrayAttr parentStaticParams = parentOp.getStaticParamsAttr();
+        ArrayAttr currentStaticParams = op.getStaticParamsAttr();
+
+        bool parentHasStatic = parentStaticParams && parentStaticParams.size() == 1;
+        bool currentHasStatic = currentStaticParams && currentStaticParams.size() == 1;
+
         auto parentTheta = parentOp.getTheta();
         auto theta = op.getTheta();
 
-        mlir::Value sumParam = rewriter.create<arith::AddFOp>(loc, parentTheta, theta).getResult();
+        mlir::Value sumParam = mlir::Value();
+        ArrayAttr SumAttr = nullptr;
+
+        if (parentHasStatic && currentHasStatic) {
+            double pVal = cast<FloatAttr>(parentStaticParams[0]).getValueAsDouble();
+            double cVal = cast<FloatAttr>(currentStaticParams[0]).getValueAsDouble();
+            SumAttr = rewriter.getArrayAttr({rewriter.getF64FloatAttr(pVal + cVal)});
+        }
+        else if (!parentHasStatic && !currentHasStatic) {
+            sumParam = rewriter.create<arith::AddFOp>(loc, parentTheta, theta);
+        }
+        else {
+            return failure();
```

Comment:
I think this case is possible to determine, but this is may be good for a different ticket.

Reply:
yes you are right! We just need to find which one is static and add a const instruction that will be used as one of the addFop arguments. I'll leave it for another PR in favour of not complicating this one too much.